src/concrete/ml/deployment/fhe_client_server.py

```diff
@@ -124,37 +124,68 @@ def load(self):
+                Tuple[fhe.Value, ...]]): The encrypted and quantized values to consider. If these
+                values are serialized (in bytes), they are first deserialized.
+            serialized_evaluation_keys (bytes): The evaluation keys. If they are serialized (in
+                bytes), they are first deserialized.
 
         Returns:
-            Union[bytes, Tuple[bytes, ...]]: the result of the model
+            Union[bytes, fhe.Value, Tuple[bytes, ...], Tuple[fhe.Value, ...]]: The model's encrypted
+                and quantized results. If the inputs were initially serialized, the outputs are also
+                serialized.
         """
+
+        # TODO: make desr / ser optional
         assert_true(self.server is not None, "Model has not been loaded.")
 
-        serialized_encrypted_quantized_data = to_tuple(serialized_encrypted_quantized_data)
+        q_data_enc = to_tuple(serialized_encrypted_quantized_data)
 
-        deserialized_data = tuple(
-            fhe.Value.deserialize(data) for data in serialized_encrypted_quantized_data
-        )
-        deserialized_keys = fhe.EvaluationKeys.deserialize(serialized_evaluation_keys)
+        # Make sure no inputs are None, to avoid any crash in Concrete
+        assert not any(x is None for x in q_data_enc), "No input values should be None"
 
-        result = self.server.run(*deserialized_data, evaluation_keys=deserialized_keys)
+        inputs_are_serialized = all(isinstance(x, bytes) for x in q_data_enc)
+        inputs_are_encrypted_values = all(isinstance(x, fhe.Value) for x in q_data_enc)
 
-        return (
-            tuple(res.serialize() for res in result)
-            if isinstance(result, tuple)
-            else result.serialize()
-        )
+        # Make sure inputs are either only serialized values or encrypted values
+        assert (
+            inputs_are_serialized ^ inputs_are_encrypted_values
+        ), "Inputs must be all of the same types, either 'bytes' or 'concrete.fhe.Value'"
+
+        # Deserialize the values if they are all serialized
+        if inputs_are_serialized:
+            q_data_enc = to_tuple(deserialize_encrypted_values(*q_data_enc))
+
+        # Deserialize the evaluation keys if they are serialized
+        evaluation_keys = serialized_evaluation_keys
+        if isinstance(evaluation_keys, bytes):
+            evaluation_keys = fhe.EvaluationKeys.deserialize(evaluation_keys)
+
+        q_result_enc = self.server.run(*q_data_enc, evaluation_keys=evaluation_keys)
+
+        # If inputs were serialized, return serialized values as well
+        if inputs_are_serialized:
+            q_result_enc = serialize_encrypted_values(*to_tuple(q_result_enc))
+
+        # Mypy complains because the outputs of `serialize_encrypted_values` can be None, but here
+        # we already made sure this is not the case
+        return q_result_enc  # type: ignore[return-value]
```

Comment:
We usually do `assert q_result_enc is not None` for mypy instead of having type: ignore

Reply:
here q_result_enc can be a tuple , and when you do something `not  all(x is None for x in q_data_enc)`, mypy still complains, but maybe there's a better way ?

Reply:
I see. I don't have any solution. mypy is boring..